### PR TITLE
Add support for Caps and Num Lock indicator

### PR DESCRIPTION
**Pull request by raibtoffoletto**
Requires https://github.com/elementary/wingpanel-indicator-keyboard/pull/62

As in the suggestion on issue wingpanel-indicator-keyboard/issues/25,
this adds a simple indicator to wingpanel showing the current status of the Caps and Num Lock keys,
with the option to activate/deactivate the indicator from the switchboard's keyboard page.

### CODE REVIEW

**Comment by ryonakano** [comment]
Could you revert changes for translation files? They should be updated automatically when this PR gets merged.

**Comment by raibtoffoletto** [comment]
Sorry about that! They are reverted now. =D

**Comment by ryonakano** [comment]
It still seems to have some conflicts.

File: src/Views/Layout.vala [hunk 1]
```diff
@@ -375,6 +386,19 @@ namespace Pantheon.Keyboard.LayoutPage {
...
+        private class StatusIndicatorSwitch : Gtk.Switch {
+            public StatusIndicatorSwitch (string key_indicator) {
+                var indicator_settings = new GLib.Settings ("io.elementary.wingpanel.keyboard-status");
+                halign = Gtk.Align.START;
+                valign = Gtk.Align.CENTER;
+                active = indicator_settings.get_boolean (key_indicator);
+
+                notify["active"].connect (() => {
+                    indicator_settings.set_boolean (key_indicator, this.active);
+                });
```
Comment (ryonakano) [inline src/Views/Layout.vala]:
```diff
-                active = indicator_settings.get_boolean (key_indicator);
-
-                notify["active"].connect (() => {
-                    indicator_settings.set_boolean (key_indicator, this.active);
-                });
+                indicator_settings.bind (key_indicator, this, "active", SettingsBindFlags.DEFAULT);
```

I guess this would be even cleaner.

Comment (raibtoffoletto) [reply src/Views/Layout.vala]: I didn't know about bind(), yes this is way better.

**Comment by raibtoffoletto** [comment]
I'm failing in finding a way check if the Caps or Num Lock modifiers exists. It seems that Gdk is able to get these status even if a physical key isn't present.
My laptop doesn't have a numpad, so no num lock, and yet gdk get the status as always active.

So I'm not sure how to perform check before packing these switches...

**Comment by cassidyjames** [comment]
@raibtoffoletto ah I can see how that comment from @danrabbit could be confusing. He meant this code should check if the GSettings keys exist, since the GSettings keys are not controlled by this package.

For example, in Ephemeral, I do a null check here:

https://github.com/cassidyjames/ephemeral/blob/d5d701e9b6e83798774dbc7ec3821318132b9e50/src/InfoBars/PaidInfoBar.vala#L86-L95

File: src/Views/Layout.vala [hunk 2]
```diff
@@ -375,6 +386,19 @@ namespace Pantheon.Keyboard.LayoutPage {
             }
         }
 
+        private class StatusIndicatorSwitch : Gtk.Switch {
+            public StatusIndicatorSwitch (string key_indicator) {
+                var indicator_settings = new GLib.Settings ("io.elementary.wingpanel.keyboard-status");
```
Comment (cassidyjames) [inline src/Views/Layout.vala]: Looks like this changes in the WingPanel branch

```suggestion
                var indicator_settings = new GLib.Settings ("io.elementary.wingpanel.keyboard");
```

**Comment by raibtoffoletto** [comment]
Oh, I see now! I was already wondering about how to check for the physical keys to not clutter the keyboard options and I misread Dan's comment.

**Comment by raibtoffoletto** [comment]
I reworked the code a bit, now it checks for a GSchema before packing the switches and labels for the indicator. Let me know if anything can be improved =D

**Comment by danrabbit** [comment]
Merged in a branch based on this one but with a few tweaks. Thanks very much for getting this started! Closing this PR since its contents are already merged

**Pull request closed without merging**